### PR TITLE
feat: capped minter v2 minting access control hook

### PR DESCRIPTION
**Pull request by marcomariscal**

### CODE REVIEW

File: l2-contracts/src/ZkCappedMinterV2.sol [hunk 1]
```diff
@@ -81,6 +91,15 @@ contract ZkCappedMinterV2 is AccessControl, Pausable {
     _grantRole(PAUSER_ROLE, _admin);
   }
 
+  /// @notice Sets the mint hook contract
+  /// @param _hook The new hook contract address, or address(0) to remove the hook
+  /// @dev Only callable by addresses with the DEFAULT_ADMIN_ROLE
+  function setMintHook(IMintHook _hook) external {
```
Comment (alexkeating) [inline l2-contracts/src/ZkCappedMinterV2.sol:97]: I was thinking the hook was set on deployment and was immutable

File: l2-contracts/src/ZkCappedMinterV2.sol [hunk 2]
```diff
@@ -109,6 +128,12 @@ contract ZkCappedMinterV2 is AccessControl, Pausable {
     _requireNotPaused();
     _checkRole(MINTER_ROLE, msg.sender);
     _revertIfCapExceeded(_amount);
+
+    // Execute hook if it exists
+    if (address(mintHook) != address(0)) {
+      mintHook.beforeMint(msg.sender, _to, _amount);
```
Comment (alexkeating) [inline l2-contracts/src/ZkCappedMinterV2.sol:134]: We should prevent the 0 address in the constructor

Comment (alexkeating) [reply l2-contracts/src/ZkCappedMinterV2.sol:134]: Open to changing the names

**Comment by marcomariscal** [comment]
closing as this is not needed anymore

**Pull request closed without merging**